### PR TITLE
add <T> through accounts index

### PR DESCRIPTION
**Pull request by jeffwashington**
#### Summary of Changes

Fixes #

### CODE REVIEW

File: runtime/src/accounts_index.rs [hunk 1]
```diff
@@ -48,7 +48,7 @@ pub type AccountMap<V> = InMemAccountsIndex<V>;
 
 pub(crate) type AccountMapEntry<T> = Arc<AccountMapEntryInner<T>>;
 
-pub trait IsCached: 'static + Clone + Debug + PartialEq + ZeroLamport + Copy + Default {
+pub trait IsCached: 'static + Clone + Debug + PartialEq + ZeroLamport + Copy + Default + Sync + Send {
```
Comment (brooksprumo) [inline runtime/src/accounts_index.rs:51]: This trait has grown quite a bit. Maybe in the next round `IsCached` can go back to just the `is_cached()` function, and a new trait can be added for this composite?

**Review by brooksprumo — COMMENTED** [review]
Can you re-request the review once this PR is rebased on top of the 'per bin stats' commit?

File: runtime/src/accounts_index.rs (hunk 1 above)
Comment (jeffwashington) [reply runtime/src/accounts_index.rs:51]: sure, seems reasonable

**Review by brooksprumo — APPROVED** [review]
lgtm

**Pull request merged**